### PR TITLE
raftstore: add several util functions

### PR DESCRIPTION
**Pull request by BusyJay**
These functions will be used in merge later.

### CODE REVIEW

File: src/raftstore/store/util.rs [hunk 1]
```diff
@@ -659,4 +700,62 @@ mod tests {
...
+
+    fn split(mut r: metapb::Region, key: &[u8]) -> (metapb::Region, metapb::Region) {
+        let mut r2 = r.clone();
+        r.set_end_key(key.to_owned());
+        r2.set_id(r.get_id() + 1);
+        r2.set_start_key(key.to_owned());
+        (r, r2)
+    }
+
+    macro_rules! check_sibling {
```
Comment (overvenus) [inline src/raftstore/store/util.rs]: Can't it be a function?

File: src/raftstore/store/util.rs [hunk 2]
```diff
@@ -323,6 +338,32 @@ impl fmt::Debug for Lease {
     }
 }
 
+/// Parse data of entry `index`.
+// TODO: make sure received entries are not corrupted
+// If this happens, TiKV will panic and can't recover without extra effort.
```
Comment (overvenus) [inline src/raftstore/store/util.rs]: Could you add a panic section? https://doc.rust-lang.org/book/first-edition/documentation.html#special-sections

**Comment by BusyJay** [comment]
ping, /run-integration-tests

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**